docs/configuration.md

```diff
@@ -358,25 +744,112 @@ http:
 		addr: localhost:5001
 ```
 
-The http option details the configuration for the HTTP server that hosts the registry.
+The `http` option details the configuration for the HTTP server that hosts the registry.
+
+<table>
+  <tr>
+    <th>Parameter</th>
+    <th>Required</th>
+    <th>Description</th>
+  </tr>
+  <tr>
+    <td>
+      <code>addr</code>
+    </td>
+    <td>
+      yes
+    </td>
+    <td>
+      The <code>HOST:PORT</code> for which the server should accept connections.
+    </td>
+  </tr>
+    <tr>
+    <td>
+      <code>prefix</code>
+    </td>
+    <td>
+      no
+    </td>
+    <td>
+If the server does not run at the root path use this value to specify the
+prefix. The root path is the section before <code>v2</code>. It
+should have both preceding and trailing slashes, for example <code>/path/</code>.
+    </td>
+  </tr>
+  <tr>
+    <td>
+      <code>secret</code>
+    </td>
+    <td>
+      yes
+    </td>
+    <td>
+A random piece of data. This is used to sign state that may be stored with the
+client to protect against tampering. For production environments you should generate a
+random piece of data using a cryptographically secure random generator.
+    </td>
+  </tr>
+</table>
 
-- addr: **Required** - The HOST:PORT for which the server should accept connections.
-- prefix: **Optional** - If the server will not run at the root path, this should specify the prefix (the part of the path before ```v2```). It should have both preceding and trailing slashes.
-- secret: A random piece of data. It is used to sign state that may be stored with the client to protect against tampering. For production use you should generate a random piece of data using a cryptographically secure random generator.
 
 ### tls
 
-The tls option within http is **optional** and allows you to configure SSL for the server. If you already have a server such as Nginx or Apache running on the same host as the registry, you may prefer to configure SSL termination there and proxy connections to the registry server.
+The `tls` struct within `http` is **optional**. Use this to configure SSL
```

Comment:
Not sure if this can fixed now: SSL and TLS seemed to be used interchangeably. We should only use TLS.